src/System.Text.Json/tests/Serialization/DictionaryTests.KeyPolicy.cs

```diff
@@ -57,19 +84,77 @@ public static void CamelCaseSerialize()
             Dictionary<string, int>[] obj = new Dictionary<string, int>[]
             {
                 new Dictionary<string, int>() { { "Key1", 1 }, { "Key2", 2 } },
-                new Dictionary<string, int>() { { "Key1", 3 }, { "Key2", 4 } },
             };
 
-            const string Json = @"[{""Key1"":1,""Key2"":2},{""Key1"":3,""Key2"":4}]";
-            const string JsonCamel = @"[{""key1"":1,""key2"":2},{""key1"":3,""key2"":4}]";
+            const string Json = @"[{""Key1"":1,""Key2"":2}]";
+            const string JsonReversed = @"[{""Key2"":2,""Key1"":1}]";
+
+            const string JsonCamel = @"[{""key1"":1,""key2"":2}]";
+            const string JsonCamelReversed = @"[{""key2"":2,""key1"":1}]";
 
             // Without key policy option, serialize keys as they are.
             string json = JsonSerializer.Serialize<object>(obj);
-            Assert.Equal(Json, json);
+            Assert.True(Json == json || JsonReversed == json);
```

Comment:
I modified the write tests to account for undetermined reflection order on serialize.

Reply:
In this case, the keys should always be in the same order. Reflection order only applies to real properties, not collection elements.

Reply:
Got it, thanks!